bridges/snowbridge/primitives/core/src/reward.rs

```diff
@@ -18,6 +19,19 @@ use xcm::{
 	prelude::{ExecuteXcm, Junction::*, Location, SendXcm, *},
 };
 
+#[derive(Debug, Clone, PartialEq, Encode, Decode, DecodeWithMemTracking, TypeInfo)]
+pub enum MessageId {
+	/// Message from Ethereum
+	Inbound(u64),
+	/// Message to Ethereum
+	Outbound(u64),
+}
+
+pub trait AddTip {
+	/// Add a relayer reward tip to a pallet.
+	fn add_tip(nonce: u64, amount: u128);
```

Comment:
If there is any possibility that we can't add a tip, for example if the nonce has already been consumed, then an error should be returned like `AddTipError::NonceConsumed`.

```suggestion
	fn add_tip(nonce: u64, amount: u128) -> Result<(), AddTipError>;
```

If the tip can't be added, then the system-v2 should track the tip so that the user can reclaim it in a future iteration of the bridge. For example, storing it in this structure:

```
LostTips[user] = LostTips[user] + tip
```